File: qb-engineer-server/qb-engineer.core/Models/CreateRecurringExpenseRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record CreateRecurringExpenseRequestModel(
    decimal Amount,
    string Category,
    string Classification,
    string Description,
    string? Vendor,
    RecurrenceFrequency Frequency,
    DateTimeOffset StartDate,
    DateTimeOffset? EndDate,
    bool AutoApprove);

File: qb-engineer-ui/src/app/features/expenses/models/create-recurring-expense-request.model.ts
import { RecurrenceFrequency } from './recurrence-frequency.type';

export interface CreateRecurringExpenseRequest {
  amount: number;
  category: string;
  classification: string;
  description: string;
  vendor?: string;
  frequency: RecurrenceFrequency;
  startDate: string;
  endDate?: string;
  autoApprove: boolean;
}
